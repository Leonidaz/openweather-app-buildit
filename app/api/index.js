import { fromJS } from 'immutable';

const owpApiKey = 'c76d6b63065dc8fc76ec535574f84631';

export const fetchWeatherData5 = () =>
  fetch(
    `http://api.openweathermap.org/data/2.5/forecast/daily?q=New%20York,us&mode=json&units=metric&APPID=${owpApiKey}`
  )
    .then(response => response.json())
    .then(response => fromJS(response))
    .catch(() =>
      fromJS({
        error: 'Something went wrong with getting a list of products',
      })
    );
