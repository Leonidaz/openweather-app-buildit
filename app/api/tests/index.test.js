import { weatherData } from 'api/tests/mockWeatherData';
import * as api from 'api';

describe('fetchWeatherData5 api call', () => {
  afterEach(() => {
    fetch.resetMocks();
  });

  it('should select the full weather state', () => {
    fetch.mockResponse(JSON.stringify(weatherData));

    return api.fetchWeatherData5().then(response => {
      expect(response).toMatchSnapshot();
    });
  });
});
