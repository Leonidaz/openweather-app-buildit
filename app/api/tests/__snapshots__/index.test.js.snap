exports[`fetchWeatherData5 api call should select the full weather state 1`] = `
Object {
  "city": Object {
    "coord": Object {
      "lat": 40.7143,
      "lon": -74.006,
    },
    "country": "US",
    "id": 5128581,
    "name": "New York",
    "population": 0,
  },
  "cnt": 7,
  "cod": "200",
  "list": Array [
    Object {
      "clouds": 0,
      "deg": 151,
      "dt": 1495382400,
      "humidity": 51,
      "pressure": 1038.38,
      "speed": 3.67,
      "temp": Object {
        "day": 19.24,
        "eve": 18.15,
        "max": 19.24,
        "min": 13.51,
        "morn": 19.24,
        "night": 13.51,
      },
      "weather": Array [
        Object {
          "description": "sky is clear",
          "icon": "01d",
          "id": 800,
          "main": "Clear",
        },
      ],
    },
    Object {
      "clouds": 92,
      "deg": 116,
      "dt": 1495468800,
      "humidity": 99,
      "pressure": 1030.73,
      "rain": 16.23,
      "speed": 3.96,
      "temp": Object {
        "day": 13.17,
        "eve": 16.14,
        "max": 16.14,
        "min": 12.23,
        "morn": 13.07,
        "night": 15.24,
      },
      "weather": Array [
        Object {
          "description": "heavy intensity rain",
          "icon": "10d",
          "id": 502,
          "main": "Rain",
        },
      ],
    },
    Object {
      "clouds": 92,
      "deg": 35,
      "dt": 1495555200,
      "humidity": 95,
      "pressure": 1023.72,
      "rain": 0.54,
      "speed": 2.02,
      "temp": Object {
        "day": 16.34,
        "eve": 16.85,
        "max": 16.85,
        "min": 14.67,
        "morn": 14.67,
        "night": 15.2,
      },
      "weather": Array [
        Object {
          "description": "light rain",
          "icon": "10d",
          "id": 500,
          "main": "Rain",
        },
      ],
    },
    Object {
      "clouds": 90,
      "deg": 171,
      "dt": 1495641600,
      "humidity": 0,
      "pressure": 999.31,
      "rain": 13.3,
      "speed": 5.35,
      "temp": Object {
        "day": 19.47,
        "eve": 19.24,
        "max": 19.47,
        "min": 18.46,
        "morn": 18.46,
        "night": 19.2,
      },
      "weather": Array [
        Object {
          "description": "heavy intensity rain",
          "icon": "10d",
          "id": 502,
          "main": "Rain",
        },
      ],
    },
    Object {
      "clouds": 51,
      "deg": 223,
      "dt": 1495728000,
      "humidity": 0,
      "pressure": 1000.74,
      "rain": 1.9,
      "speed": 5.04,
      "temp": Object {
        "day": 20.68,
        "eve": 18.9,
        "max": 20.68,
        "min": 15.29,
        "morn": 18.11,
        "night": 15.29,
      },
      "weather": Array [
        Object {
          "description": "light rain",
          "icon": "10d",
          "id": 500,
          "main": "Rain",
        },
      ],
    },
    Object {
      "clouds": 47,
      "deg": 286,
      "dt": 1495814400,
      "humidity": 0,
      "pressure": 1008.35,
      "rain": 0.38,
      "speed": 5.14,
      "temp": Object {
        "day": 17.49,
        "eve": 15.83,
        "max": 17.49,
        "min": 12.52,
        "morn": 14.86,
        "night": 12.52,
      },
      "weather": Array [
        Object {
          "description": "light rain",
          "icon": "10d",
          "id": 500,
          "main": "Rain",
        },
      ],
    },
    Object {
      "clouds": 13,
      "deg": 249,
      "dt": 1495900800,
      "humidity": 0,
      "pressure": 1017.58,
      "rain": 1.3,
      "speed": 2.25,
      "temp": Object {
        "day": 19.33,
        "eve": 17.18,
        "max": 19.33,
        "min": 12.07,
        "morn": 14,
        "night": 12.07,
      },
      "weather": Array [
        Object {
          "description": "light rain",
          "icon": "10d",
          "id": 500,
          "main": "Rain",
        },
      ],
    },
  ],
  "message": 0.4449659,
}
`;
